### DISCUSSION TITLE
Using application/x-www-form-urlencoded in headers and body of RestAdvanced doesnt work

### DISCUSSION THREAD

**Discussion by nikos445** [ansibleguy76/ansibleforms#180] [General]
**Describe the bug**
Using **application/x-www-form-urlencoded** in **headers** and **body** doesn't **encode in URL format** the body or headers

**To Reproduce**

> expression: fn.fnRestAdvanced(
'post',
$(test.host)/login',
'username=$(test.user)&password=$(test.password)&client_id=$(test.user)&client_secret=$(test.password)',
{'Content-Type':'application/x-www-form-urlencoded;charset=UTF-8'},
'.access_token',
'')

**Expected behavior**
URL encode the user that has domain\username
URL encode the password that has symbols %^'/*!&#\ etc...

**Screenshots**
N/A 

**Version**
ansibleforms v5.0.1

**Deployment**
Deployed ansibleforms with :
- nodejs native

**Additional context**

**Comment by ansibleguy76** [comment]
hi

if you want to have the headers encoded in a specific way, you should encode them yourself.  make a hidden intermediate expression and encode as you wish (runLocal true will allow the full javascript engine) and then use the outcome of that intermediate field in your custom headers.

**Comment by nikos445** [comment]
Oki but I think that as long as you specify this header (application/x-www-form-urlencoded) it shall be done automatically...

**Reply by ansibleguy76** [reply]
can you elaborate, I seem to be missing something.

**Reply by ansibleguy76** [reply]
I'm just calling rest api's.  If you want to go advanced and use special headers, you should add them accordingly.  

I basically do :

```
    if(action=="get"){
      data = await axios.get(url,axiosConfig);
    }else if(action=="post"){
      data = await axios.post(url,body,axiosConfig);
    }else if(action=="delete"){
      data = await axios.post(url,axiosConfig);
    }else if(action=="put"){
      data = await axios.put(url,body,axiosConfig);
    }
```

and axiosConfig.headers will contain the headers, which are derived directly from what you says they should be.

**Reply by nikos445** [reply]
Oki I understand... Then maybe its better to add in documentation an example for url encoding... cause its hard to debug when you dont know why its not working... 
So this way you dont need to modify code ;-)

**Reply by ansibleguy76** [reply]
would you be so kind to provide a valid example, this way i know for sure, i got it right.

**Comment by ansibleguy76** [comment]
closing as stale.
